extension/src/pipeline/quickPick.ts

```diff
@@ -61,109 +68,160 @@ const pickTemplateAndFields = async (
   return { template, x, y }
 }
 
-export type PlotConfigData = {
-  x: { file: string; key: string }
-  template: string
-  y: { file: string; key: string }
+const joinList = (items: string[]) => {
+  if (items.length <= 2) {
+    return items.join(' and ')
+  }
+
+  return `${items.slice(0, -1).join(', ')}, and ${items[items.length - 1]}`
 }
 
-type UnknownValue = Value | ValueTree
+const validateFileNames = (files: string[] | undefined) => {
+  if (!files) {
+    return []
+  }
+  const fileExts = [...new Set(files.map(file => getFileExtension(file)))]
 
-const getFieldsFromArr = (dataArr: UnknownValue[]) => {
+  if (fileExts.length > 1) {
+    void Toast.showError(
+      `Found files with ${joinList(
+        fileExts
+      )} extensions. Files must be of the same type.`
+    )
+    return []
+  }
+  return files
+}
+
+const getMetricInfoFromArr = (
+  dataArr: UnknownValue[]
+): { arrLength: number; fields: string[] } | undefined => {
   const firstArrVal: UnknownValue = dataArr[0]
   if (!isValueTree(firstArrVal)) {
-    return []
+    return
   }
   const fieldObjKeys = Object.keys(firstArrVal)
   const objsHaveSameKeys = dataArr.every(
     val => isValueTree(val) && isEqual(fieldObjKeys, Object.keys(val))
   )
-  return objsHaveSameKeys ? fieldObjKeys : []
+  if (!objsHaveSameKeys) {
+    return
+  }
+  return { arrLength: dataArr.length, fields: fieldObjKeys }
 }
 
-const getFieldsFromValue = (data: UnknownValue): string[] => {
+const getMetricInfoFromValue = (
```

Comment:
Instead of just fields, we need to get the length of the array found inside of the file data as well. I renamed the function from `getFieldsFromValue` to `getMetricInfoFromValue` since `getFieldsAndArrLengthFromValue` looked pretty lengthy :)